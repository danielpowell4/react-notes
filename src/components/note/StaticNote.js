import React from 'react';
import {partial} from '../../lib/utils'

export const StaticNote = (props) => {
  const handleRemove = partial(props.handleRemove, props.id)
  const handleToggleEdit = partial(props.handleToggleEdit, props.id)

  return (
    <li>
      <span className='delete-item'><a href="#" onClick={handleRemove}>X</a></span>
      <span onClick={handleToggleEdit}>{props.content}</span>
    </li>
  )
}

  StaticNote.propTypes = {
    content: React.PropTypes.string.isRequired,
    handleToggleEdit: React.PropTypes.func.isRequired,
    handleRemove: React.PropTypes.func.isRequired,
    id: React.PropTypes.number.isRequired
  }
