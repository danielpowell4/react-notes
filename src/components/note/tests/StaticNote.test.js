import React from 'react';
import renderer from 'react-test-renderer';
import {StaticNote} from '../StaticNote';

/* StaticNote snapshot test */

const staticNoteSampleProps = { content: 'static note',
                                handleToggleEdit: function(){},
                                handleRemove: function(){},
                                id: 1 }

test('StaticNote renders correctly', () => {
  const tree = renderer.create(
    <StaticNote {...staticNoteSampleProps} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
