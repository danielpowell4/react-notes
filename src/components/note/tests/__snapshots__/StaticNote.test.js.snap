exports[`test StaticNote renders correctly 1`] = `
<li>
  <span
    className="delete-item">
    <a
      href="#"
      onClick={[Function]}>
      X
    </a>
  </span>
  <span
    onClick={[Function]}>
    static note
  </span>
</li>
`;
